### PR TITLE
Runchecker job optimization

### PR DESCRIPTION
**Pull request by t8m**
Put jobs that are more likely to fail to on pull request CI.
To compensate move some less likely to fail jobs to on push and daily CI jobs.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Depends on #21333

File: .github/workflows/run-checker-daily.yml [hunk 1]
```diff
@@ -116,16 +112,17 @@ jobs:
           no-ssl-trace,
           no-static-engine no-shared,
           no-stdio,
+          no-tests,
           enable-tfo,
           no-tls1,
           no-tls1_1,
           no-tls1_1-method,
-          no-tls1_2,
```
Comment (tom-cosgrove-arm) [inline .github/workflows/run-checker-daily.yml:123]: this doesn't seem to get added anywhere - is that okay?

File: .github/workflows/run-checker-ci.yml [hunk 2]
```diff
@@ -19,26 +19,23 @@ jobs:
...
           no-cms,
-          no-ct,
+          no-dgram,
+          no-dh,
           no-dtls,
           no-ec,
           no-ec2m,
+          no-ecx,
           no-http,
-          no-siv,
```
[the remark this answers is not in the document]
Comment (t8m) [reply .github/workflows/run-checker-ci.yml:27]: This one is already in run-checker-daily.yml

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to master branch. Thank you for the reviews.

**Pull request closed without merging**